ee/bulk-operations-dashboard/api/controllers/software/edit-software.js

```diff
@@ -0,0 +1,343 @@
+        // for(let team of addedTeams) {
+        await sails.helpers.flow.simultaneouslyForEach(addedTeams, async (team)=>{
+          // console.log(`transfering ${software.name} to fleet instance for team id ${team}`);
+          // Send an api request to send the file to the Fleet server for each added team.
+          await sails.cp(softwareFd, {},
+            {
+              adapter: ()=>{
+                return {
+                  ls: undefined,
+                  rm: undefined,
+                  read: undefined,
+                  receive: (unusedOpts)=>{
+                    // This `_write` method is invoked each time a new file is received
+                    // from the Readable stream (Upstream) which is pumping filestreams
+                    // into this receiver.  (filename === `__newFile.filename`).
+                    var receiver__ = WritableStream({ objectMode: true });
+                    // Create a new drain (writable stream) to send through the individual bytes of this file.
+                    receiver__._write = (__newFile, encoding, doneWithThisFile)=>{
+
+                      let FormData = require('form-data');
+                      let form = new FormData();
+                      form.append('team_id', team);
+                      form.append('install_script', installScript);
+                      form.append('post_install_script', postInstallScript);
+                      form.append('pre_install_query', preInstallQuery);
+                      form.append('uninstall_script', uninstallScript);
+                      form.append('software', __newFile, {
+                        filename: software.name,
+                        contentType: 'application/octet-stream'
+                      });
+                      (async ()=>{
+                        try {
+                          await axios.post(`${sails.config.custom.fleetBaseUrl}/api/v1/fleet/software/package`, form, {
+                            headers: {
+                              Authorization: `Bearer ${sails.config.custom.fleetApiToken}`,
+                              ...form.getHeaders()
+                            },
+                          });
+                        } catch(error){
+                          throw new Error('Failed to upload file:'+ require('util').inspect(error, {depth: null}));
+                        }
+                      })()
+                      .then(()=>{
+                        // console.log('ok supposedly a file is finished uploading');
+                        doneWithThisFile();
+                      })
+                      .catch((err)=>{
+                        console.log(err);
+                        doneWithThisFile(err);
+                      });
+                    };//ƒ
+                    return receiver__;
+                  }
+                };
+              },
+            })
+          .intercept((unusedErr)=>{
+            return 'softwareUploadFailed';
```

Comment:
Rename unusedErr +
```suggestion
            return {'softwareUploadFailed': unusedErr};
```

Reply:
Add a comment about this error catching every error. TODO: add a note about future negotiation of errors and 
1. that I did not test the behavior of uploading a software to a team it was already deployed to to verify that it returns a 409 response.
2. That it currently swallows all errors and returns a softwareUploadFailed response.